addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -66,6 +67,42 @@ class HrWorkEntry(models.Model):
     def init(self):
         tools.create_index(self._cr, "hr_work_entry_date_start_date_stop_index", self._table, ["date_start", "date_stop"])
 
+    def _get_work_entries_data(self, work_entries_dict):
+        data = self.env['hr.work.entry']
+        for key, work_entries in work_entries_dict.items():
+            actual_duration = sum(work_entry.duration for work_entry in work_entries)
+            week_type = "0" if key[0].isocalendar()[1] % 2 == 0 else "1"
+            filter_condition = lambda x: (not key[1].resource_calendar_id.two_weeks_calendar or (x.week_type == week_type)) and int(x.dayofweek) == key[0].weekday() and x.day_period != 'lunch'
+            total_work_entries = key[1].resource_calendar_id.attendance_ids.filtered(filter_condition)
+            total_duration = round(sum(work_entry.duration_hours for work_entry in total_work_entries), 2)
+            if actual_duration < total_duration:
+                data |= work_entries
+        return data
```

Comment:
I should have noticed it earlier, but this method and the override are very similar and changing something here means modifying the other. It would be better to split this method with submethods like `_get_actual_duration` on the work entry model and `_get_calendar` on the employee model and override those in  the calendar bridges instead :) That way `_get_work_entries_data` won't need to be overriden.

Reply:
Hello @Bertrand2 

Me too :slightly_smiling_face: 
I should've split the code into the methods instead of overring the whole method :pray: 
Thank you for looking into it